### DISCUSSION TITLE
Cannot set local name and location

### DISCUSSION THREAD

**Discussion by spudwebb** [zwave-js/ZWaveJS.NET#60] [General]
I'm trying to set the local name and location of a node from my application. Basically the fields you see in ZWave JS UI in the screenshot below
![image](https://github.com/zwave-js/ZWaveJS.NET/assets/5427056/1831ff29-436d-415d-99b0-b66d6919089f)

For that, I'm using the` ZWaveNode.SetName()` and `ZWaveNode.SetLocation()` methods from ZWaveJS.NET, but those calls seem to always try to update the name and location of the "Node Naming and Location" command class  even if the device does not supports it, and even if the updateCC parameter is false, and more importantly they never update the local name and location properties.

I don't know where the problem is exactly, any help is appreciated.

**Comment by marcus-j-davies** [comment]
Hi @spudwebb 

The `Name` & `Location` is.... funny.

> but those calls seem to always try to update the name and location of the "Node Naming and Location" command class even if the device does not supports it, and even if the updateCC parameter is false

The driver itself has a 'local only' cache of a node' name and location, and will survive a reboot of the Node JS runtime (given its written to cache)

So these values get updated regardless, even if setting `updateCC` to `false`

HOWEVER, it will attempt to 'burn' (is that the right word) these values to the device itself if it supports the ability to hard write the values to the Devices memory -  if setting `updateCC` to `true`

You can see this check happening here in the Server that the .NET lib connects to:

https://github.com/zwave-js/zwave-js-server/blob/81b05b604ecfab5e09c0926bf0ce3fc755891b2f/src/lib/node/message_handler.ts#L245

> and more importantly they never update the local name and location properties

I am more interested in this  - The .NET Class instance of a Node should be updated (providing the the call to the server is successful)

https://github.com/zwave-js/ZWaveJS.NET/blob/6c3db915019895bd99a28edd7370bec5866ecfdb/Visual%20Studio%20Projects/ZWaveJS.NET/ZWaveJS.NET/ZWaveNode.cs#L880

https://github.com/zwave-js/ZWaveJS.NET/blob/6c3db915019895bd99a28edd7370bec5866ecfdb/Visual%20Studio%20Projects/ZWaveJS.NET/ZWaveJS.NET/CMDResult.cs#L15

can you trace this call - to see if its successful?

In theory there are these copies.

 - The Zwave JS Local Cache version (updates regardless)
 - The .NET runtime instance value (updates if a call to update is successfull)
 - The Device's hard/burned writren value (if supported + UpdateCC is set to true)

**Comment by spudwebb** [comment]
Hi @marcus-j-davies,
Yes the call is successful, and I can see that the .NET runtime instance value is updated and it is also being written to my store/*.values.jsonl file. So everything works as expected in ZWaveJS.NET and ZWave JS Server.

I now realize that my problem is specifically with ZWave JS UI, I expected my update calls to also update the name and location that  are displayed in the ZWave JS UI page, but it seems that ZWave JS UI uses its own instance of name and locations that are saved in store/nodes.json and they do not use the ZWave JS Local cache version to populate those fields.

I will create an issue in the ZWave JS UI project.

**Comment by spudwebb** [comment]
This issue has been fixed in ZWave JS UI, see https://github.com/zwave-js/zwave-js-ui/issues/3591

**Reply by marcus-j-davies** [reply]
Nice! - sorry should have responded last time, super busy at the moment  🤪
